.gitlab/ci/on-push.yml

```diff
@@ -478,3 +478,43 @@ xsoar_ng_server_ga:
     MARKETPLACE_NAME: "xsoar"
   needs: ["xsoar-prepare-testing-bucket"]
   allow_failure: true
+
+
+test_upload_flow:
+  tags:
+    - gke
+  extends:
+    - .default-job-settings
+  rules:
+    - if: '$CI_PIPELINE_SOURCE =~ /^(push|contrib)$/'
+  variables:
+    CURRENT_BRANCH: "$CI_COMMIT_BRANCH"
+    ALL_BUCKETS: "$GCS_MARKET_BUCKET_DEV,$GCS_MARKET_V2_BUCKET_DEV"
+  stage: unittests-and-validations
+  allow_failure: true
+  script:
+    - |
+      if ! $(git diff origin/master $CI_COMMIT_BRANCH --name-only | grep -q "Tests/\|Utils/") ; then
```

Comment:
FYI this can be done also in the [rules:changes](https://docs.gitlab.com/ee/ci/yaml/#ruleschanges)